pandas/io/formats/console.py

```diff
@@ -82,3 +81,75 @@ def get_console_size():
     # and we're in a script (non-inter), this will return (None,None)
     # caller needs to deal.
     return (display_width or terminal_width, display_height or terminal_height)
+
+
+# ----------------------------------------------------------------------
+# Detect our environment
+
+def in_interactive_session():
+    """ check if we're running in an interactive shell
+
+    returns True if running under python/ipython interactive shell
+    """
+    from pandas import get_option
```

Comment:
already imported

Reply:
In a different function, not at module level

Reply:
ahh ok